File: javascript/simple-math/simple-math.js
import { input } from "pynput";

const first_number = Number(await input("What is the first number? "));
const second_number = Number(await input("What is the second number? "));

const add = (a, b) => a + b;
const subtract = (a, b) => a - b;
const multiply = (a, b) => a * b;
const divide = (a, b) => (b === 0 ? 0 : a / b);

const sum = add(first_number, second_number);
const difference = subtract(first_number, second_number);
const product = multiply(first_number, second_number);
const quotient = divide(first_number, second_number);

console.log(`${first_number} + ${second_number} = ${sum}`);
console.log(`${first_number} - ${second_number} = ${difference}`);
console.log(`${first_number} * ${second_number} = ${product}`);
console.log(`${first_number} / ${second_number} = ${quotient}`);

File: zig/simple_math.zig
const std = @import("std");

pub fn main() !void {
    const stdin = std.io.getStdIn().reader();
    var buffer: [50]u8 = undefined;

    std.debug.print("What is the first number? ", .{});
    var input = try stdin.readUntilDelimiter(buffer[0..], '\n');
    const first_number = try std.fmt.parseInt(i32, input, 10);

    std.debug.print("What is the second number? ", .{});
    input = try stdin.readUntilDelimiter(buffer[0..], '\n');
    const second_number = try std.fmt.parseInt(i32, input, 10);

    std.debug.print("{d} + {d} = {d}\n", .{ first_number, second_number, first_number + second_number });
    std.debug.print("{d} - {d} = {d}\n", .{ first_number, second_number, first_number - second_number });
    std.debug.print("{d} * {d} = {d}\n", .{ first_number, second_number, first_number * second_number });
    std.debug.print("{d} / {d} = {d}\n", .{ first_number, second_number, if (second_number <= 0) 0 else @divTrunc(first_number, second_number) });
}
